extensions/src/main/java/org/pentaho/platform/web/http/api/resources/RepositoryFileStreamProvider.java

```diff
@@ -86,6 +95,18 @@ public OutputStream getOutputStream() throws Exception {
     String extension = RepositoryFilenameUtils.getExtension( tempOutputFilePath );
     if ( "*".equals( extension ) ) { //$NON-NLS-1$
       tempOutputFilePath = tempOutputFilePath.substring( 0, tempOutputFilePath.lastIndexOf( '.' ) );
+
+      if ( appendDateFormat != null ) {
+        try {
+          Date date = new Date();
```

Comment:
Let's use the new api for date/time.  Here's a general example:

LocalDate now = LocalDate.now();DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");String formattedDate = now.format(formatter);LocalDate parsedDate = LocalDate.parse(formattedDate, formatter);
--

Reply:
@kurtwalker 
This will not work with "yyyyMMddHHmmss" format, LocaleDate does not store or represent a time or time-zone so .format throws 'UnsupportedTemporalTypeException: Unsupported field: HourOfDay'

Reply:
Will LocalTime work?

Reply:
LocalDateTime will